atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/bank/BankBalanceWorkflows.java

```diff
@@ -75,21 +72,14 @@ static Workflow create(
             AtomicBoolean skipRunning,
             Consumer<Map<Integer, Optional<Integer>>> onFailure) {
         BankBalanceRunTask runTask = new BankBalanceRunTask(skipRunning, bankBalanceWorkflowConfiguration, onFailure);
-        return DefaultWorkflow.create(
-                store,
-                (txnStore, _index) -> runTask.run(txnStore),
-                bankBalanceWorkflowConfiguration,
-                executionExecutor);
+        return DefaultWorkflow.create(store, runTask, bankBalanceWorkflowConfiguration, executionExecutor);
     }
 
-    private static class BankBalanceRunTask {
-
-        private final Consumer<Map<Integer, Optional<Integer>>> onFailure;
-        private final AtomicBoolean skipRunning;
-
+    private static class BankBalanceRunTask
+            extends StoppableKeyedTransactionTask<InteractiveTransactionStore, Map<Integer, Optional<Integer>>> {
         private final BankBalanceWorkflowConfiguration workflowConfiguration;
 
-        public BankBalanceRunTask(BankBalanceWorkflowConfiguration workflowConfiguration) {
+        private BankBalanceRunTask(BankBalanceWorkflowConfiguration workflowConfiguration) {
```

Comment:
👀 whoops, missed this on the first run (I guess it doesn't really matter since the class is private, but this is correct)